ee/localserver/krypto-unwrap_test.go

```diff
@@ -0,0 +1,165 @@
+package localserver
+
+import (
+	"bytes"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/kolide/kit/ulid"
+	"github.com/kolide/krypto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUnwrapV0(t *testing.T) {
+	t.Parallel()
+
+	myKey, err := krypto.RsaRandomKey()
+	require.NoError(t, err)
+
+	counterpartyKey, err := krypto.RsaRandomKey()
+	require.NoError(t, err)
+
+	counterpartyPub, ok := counterpartyKey.Public().(*rsa.PublicKey)
+	require.True(t, ok)
+
+	malloryKey, err := krypto.RsaRandomKey()
+	require.NoError(t, err)
+
+	counterpartyBoxer := krypto.NewBoxer(counterpartyKey, nil)
+
+	malloryBoxer := krypto.NewBoxer(malloryKey, nil)
+
+	// Make some signed boxes
+	signedIncorrectBox, err := counterpartyBoxer.Sign("", []byte("incorrect"))
+	require.NoError(t, err)
+
+	expectedCmd := ulid.New()
+	expectedId := ulid.New()
+
+	cmdReq := mustMarshal(t, cmdRequestType{Cmd: expectedCmd, Id: expectedId})
+
+	signedBox, err := counterpartyBoxer.Sign("", cmdReq)
+	require.NoError(t, err)
+
+	mallorySigned, err := malloryBoxer.Sign("", cmdReq)
+	require.NoError(t, err)
+
+	var tests = []struct {
+		name      string
+		boxParam  string
+		loggedErr string
+	}{
+		{
+			name:      "no command",
+			boxParam:  "",
+			loggedErr: "no data in box query parameter",
+		},
+		{
+			name:      "bad base64",
+			boxParam:  "This is not base64",
+			loggedErr: "unable to base64 decode box",
+		},
+
+		{
+			name:      "no signature",
+			boxParam:  "aGVsbG8gd29ybGQK",
+			loggedErr: "unable to verify box",
+		},
+
+		{
+			name:      "malformed cmd",
+			boxParam:  base64.StdEncoding.EncodeToString(signedIncorrectBox),
+			loggedErr: "unable to unmarshal cmd request",
+		},
+
+		{
+			name:      "wrong signature",
+			boxParam:  base64.StdEncoding.EncodeToString(mallorySigned),
+			loggedErr: "unable to verify box",
+		},
+
+		{
+			name:     "works",
+			boxParam: base64.StdEncoding.EncodeToString(signedBox),
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			var logBytes bytes.Buffer
+
+			kbm, err := NewKryptoBoxerMiddleware(log.NewLogfmtLogger(&logBytes), myKey, counterpartyPub)
+			require.NoError(t, err)
+
+			h := kbm.UnwrapV1Hander(makeTestHandler(t))
+			req := makeRequest(t, tt.boxParam)
+
+			rr := httptest.NewRecorder()
+			h.ServeHTTP(rr, req)
+
+			if tt.loggedErr != "" {
+				assert.Equal(t, http.StatusUnauthorized, rr.Code)
+				assert.Contains(t, logBytes.String(), tt.loggedErr)
```

Comment:
This is a really cool! There a have been times where I had a func that didn't return an error, but wanted to verify that it was logging them. This will be useful.

Reply:
I'm super pleased with myself for it.